cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java

```diff
@@ -1163,6 +1175,118 @@ protected Future<Void> clusterId(KafkaStatus kafkaStatus) {
                 });
     }
 
+    public Future<Void> canScaleDownBrokers(KafkaStatus kafkaStatus) {
+
+        if (currentReplicas != 0 &&  kafka.getReplicas() != 0 && currentReplicas > kafka.getReplicas()) {
+
+            int podId = currentReplicas - 1;
+            for (int i = 0; i < currentReplicas - kafka.getReplicas(); i++) {
+
+                canScaleDownBrokers(podId, kafkaStatus);
+
+                if (!canScaleDown) {
+                    break;
+                }
+                podId--;
+            }
+        }
+        return Future.succeededFuture();
+    }
+    public Future<Void> canScaleDownBrokers(int podId, KafkaStatus kafkaStatus) {
+
+        return ReconcilerUtils.clientSecrets(reconciliation, secretOperator)
+                .compose(compositeFuture -> {
+                    Promise<Void> resultPromise = Promise.promise();
+                    vertx.createSharedWorkerExecutor("kubernetes-ops-pool").executeBlocking(
+                            future -> {
+                                Admin kafkaAdmin = null;
+                                final Future<Collection<TopicDescription>> descriptions;
+                                try {
+                                    String bootstrapHostname = KafkaResources.bootstrapServiceName(reconciliation.name()) + "." + reconciliation.namespace() + ".svc:" + KafkaCluster.REPLICATION_PORT;
+                                    LOGGER.debugCr(reconciliation, "Creating AdminClient for clusterId using {}", bootstrapHostname);
```

Comment:
can't you log `<namespace>/<name>` to identify the cluster instead of the bootstrap?